ebpf/tracer.go

```diff
@@ -234,6 +248,8 @@ func (t *Tracer) getConnections(active []ConnectionStats) ([]ConnectionStats, ui
 			if t.shouldSkipConnection(&conn) {
 				atomic.AddUint64(&t.skippedConns, 1)
 			} else {
+				// lookup conntrack in for active
+				conn.IPTranslation = t.conntracker.GetTranslationForConn(conn.Source, conn.SPort)
```

Comment:
This still doesn't apply to short-lived connections.

I think you should move this call and `ClearShortLived()` to `GetActiveConnections()` after we've merged short-lived + active connections into a single list in 	

```
return &Connections{Conns: t.state.Connections(clientID, latestTime, latestConns)}, nil
```

Reply:
I just pushed a change to handle short lived connections in initPerfPolling.  Thinking about it some more it's still racey. 

I don't follow your suggestion. GetActiveConnections is not doing any merging. All the merging happens in `state.go`

Reply:
Yeah, it gets merged via `t.state.Connections()` which is called in `GetActiveConnections()`. You could do it all there if you wanted.

What bit is still racey?

Reply:
> Yeah, it gets merged via t.state.Connections() which is called in GetActiveConnections(). You could do it all there if you wanted.

Are you suggesting adding the translations after calling `t.state.Connections()` ? I want to merge before (or inside `t.state.Connections()`) because the entries returned by `t.state.Connections` can be older which means entries need to stick around in `conntracker` longer.  Doing it inside  `t.state.connections() ` could work, but I'd like to avoid adding more code there if possible. 

> initPerfPolling is fine too though

> What bit is still racey?

There's no guarantee that the netlink callback to initialize the conntrack entry gets called before the `initPerfPolling` callback.

Reply:
>Are you suggesting adding the translations after calling t.state.Connections() ? I want to merge before (or inside t.state.Connections()) because the entries returned by t.state.Connections can be older which means entries need to stick around in conntracker longer.

Yeah the suggestion was to add it after calling `t.state.Connections()`. Aren't you storing all translations until this point anyways? I didn't see any code that clears it earlier than `ClearShortLived`.

> Doing it inside t.state.connections() could work, but I'd like to avoid adding more code there if possible.

Yeah, I'd prefer not to have anything more in here as well.

> There's no guarantee that the netlink callback to initialize the conntrack entry gets called before the initPerfPolling callback.

While it's not great memory wise, holding on to the translations for longer reduces this risk, since you're delaying the mapping of connection to it's translation. 

It would be nice if there was a way to not depend on that though.

Reply:
Either way, I think it's more than fine to continue what you're doing here with `initPerfPolling`, though you'll have to think of something re: that race condition.